bucket_api.create_bucket fails to create bucket either nsfs/normal type

### Environment info
INFO[0000] CLI version: 5.9.0                           
INFO[0000] noobaa-image: noobaa/noobaa-core:master-20210616 [but I have seen same issue on 0622 build as well]
INFO[0000] operator-image: noobaa/noobaa-operator:5.9.0 

OCP:
Server Version: 4.7.12
Kubernetes Version: v1.20.0+df9c838


### Actual behavior
Observed that bucket_api.create_bucket fails to create buckets after user deletes all buckets and starts over with bucket creations again. 

### Expected behavior
Observed that bucket_api.create_bucket should succeed to create buckets even after user deletes all buckets and starts over with bucket creations. 

### Steps to reproduce
1. Delete all buckets which are shown under `noobaa bucket list` using `noobaa bucket delete <name>` command.
2. Attempt to create a new nsfs bucket using bucket_api.create_bucket
```
noobaa api bucket_api create_bucket '{"name":"bucket1user01","namespace":{"write_resource":{"resource":"fs1","path":"bucket1"},"read_resources":[{"resource":"fs1","path":"bucket1"}]}}'
```
or
``` 
noobaa api bucket_api create_bucket '{"name":"bucket1user01"}'
```
It fails with below error
```
FATA[0001] ❌ Cannot read property 'bucket1user01#kqbuft7c' of undefined 
```
More logs in below section.

3. However I could create buckets via the accounts created.
```
# AWS_ACCESS_KEY_ID=T65pA58Fh4na2upZoW1N AWS_SECRET_ACCESS_KEY=sT6t6CcUyCmpKtjZyqoSZSB3e+3YOE80Fyn7lMbH aws --endpoint https://s3-noobaa.apps.hpoocpci.cp.fyre.ibm.com --no-verify-ssl s3 mb s3://user01bucket
urllib3/connectionpool.py:1013: InsecureRequestWarning: Unverified HTTPS request is being made to host 's3-noobaa.apps.hpoocpci.cp.fyre.ibm.com'. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings
make_bucket: user01bucket

# noobaa bucket list
INFO[0001] ✅ Exists: NooBaa "noobaa"                    
INFO[0001] ✅ Exists: Service "noobaa-mgmt"              
INFO[0001] ✅ Exists: Secret "noobaa-operator"           
INFO[0001] ✅ Exists: Secret "noobaa-admin"              
INFO[0001] ✈️  RPC: bucket.list_buckets() Request: <nil> 
WARN[0001] RPC: GetConnection creating connection to wss://localhost:41053/rpc/ 0xc00072df40 
INFO[0001] RPC: Connecting websocket (0xc00072df40) &{RPC:0xc0004ff270 Address:wss://localhost:41053/rpc/ State:init WS:<nil> PendingRequests:map[] NextRequestID:0 Lock:{state:1 sema:0} ReconnectDelay:0s} 
INFO[0001] RPC: Connected websocket (0xc00072df40) &{RPC:0xc0004ff270 Address:wss://localhost:41053/rpc/ State:init WS:<nil> PendingRequests:map[] NextRequestID:0 Lock:{state:1 sema:0} ReconnectDelay:0s} 
INFO[0001] ✅ RPC: bucket.list_buckets() Response OK: took 0.3ms 

BUCKET-NAME    
user01bucket   

# noobaa bucket status user01bucket
INFO[0001] ✅ Exists: NooBaa "noobaa"                    
INFO[0001] ✅ Exists: Service "noobaa-mgmt"              
INFO[0001] ✅ Exists: Secret "noobaa-operator"           
INFO[0001] ✅ Exists: Secret "noobaa-admin"              
INFO[0001] ✈️  RPC: bucket.read_bucket() Request: {Name:user01bucket} 
WARN[0001] RPC: GetConnection creating connection to wss://localhost:44173/rpc/ 0xc00007ce60 
INFO[0001] RPC: Connecting websocket (0xc00007ce60) &{RPC:0xc00059c500 Address:wss://localhost:44173/rpc/ State:init WS:<nil> PendingRequests:map[] NextRequestID:0 Lock:{state:1 sema:0} ReconnectDelay:0s} 
INFO[0001] RPC: Connected websocket (0xc00007ce60) &{RPC:0xc00059c500 Address:wss://localhost:44173/rpc/ State:init WS:<nil> PendingRequests:map[] NextRequestID:0 Lock:{state:1 sema:0} ReconnectDelay:0s} 
INFO[0001] ✅ RPC: bucket.read_bucket() Response OK: took 5.3ms 

Bucket status:
  Bucket                 : user01bucket
  Type                   : NAMESPACE
  Mode                   : OPTIMAL
  Num Objects            : 0
  Data Size              : 0.000 B
  Data Size Reduced      : 0.000 B
  Data Space Avail       : 0.000 B

```
### More information - Screenshots / Logs / Other output
```
[root@hpoocpci-inf ~]# noobaa bucket list
INFO[0001] ✅ Exists: NooBaa "noobaa"                    
INFO[0001] ✅ Exists: Service "noobaa-mgmt"              
INFO[0001] ✅ Exists: Secret "noobaa-operator"           
INFO[0001] ✅ Exists: Secret "noobaa-admin"              
INFO[0001] ✈️  RPC: bucket.list_buckets() Request: <nil> 
WARN[0001] RPC: GetConnection creating connection to wss://localhost:35291/rpc/ 0xc000a6c0a0 
INFO[0001] RPC: Connecting websocket (0xc000a6c0a0) &{RPC:0xc000534500 Address:wss://localhost:35291/rpc/ State:init WS:<nil> PendingRequests:map[] NextRequestID:0 Lock:{state:1 sema:0} ReconnectDelay:0s} 
INFO[0001] RPC: Connected websocket (0xc000a6c0a0) &{RPC:0xc000534500 Address:wss://localhost:35291/rpc/ State:init WS:<nil> PendingRequests:map[] NextRequestID:0 Lock:{state:1 sema:0} ReconnectDelay:0s} 
INFO[0001] ✅ RPC: bucket.list_buckets() Response OK: took 0.3ms 
No buckets found.
[root@hpoocpci-inf ~]# noobaa api bucket_api create_bucket '{"name":"bucket1user01","namespace":{"write_resource":{"resource":"fs1","path":"bucket1"},"read_resources":[{"resource":"fs1","path":"bucket1"}]}}'
INFO[0001] ✅ Exists: NooBaa "noobaa"                    
INFO[0001] ✅ Exists: Service "noobaa-mgmt"              
INFO[0001] ✅ Exists: Secret "noobaa-operator"           
INFO[0001] ✅ Exists: Secret "noobaa-admin"              
INFO[0001] ✈️  RPC: bucket.create_bucket() Request: map[name:bucket1user01 namespace:map[read_resources:[map[path:bucket1 resource:fs1]] write_resource:map[path:bucket1 resource:fs1]]] 
WARN[0001] RPC: GetConnection creating connection to wss://localhost:34345/rpc/ 0xc00003bb30 
INFO[0001] RPC: Connecting websocket (0xc00003bb30) &{RPC:0xc00058b090 Address:wss://localhost:34345/rpc/ State:init WS:<nil> PendingRequests:map[] NextRequestID:0 Lock:{state:1 sema:0} ReconnectDelay:0s} 
INFO[0001] RPC: Connected websocket (0xc00003bb30) &{RPC:0xc00058b090 Address:wss://localhost:34345/rpc/ State:init WS:<nil> PendingRequests:map[] NextRequestID:0 Lock:{state:1 sema:0} ReconnectDelay:0s} 
ERRO[0001] ⚠️  RPC: bucket.create_bucket() Response Error: Code=INTERNAL Message=Cannot read property 'bucket1user01#kqbuft7c' of undefined 
FATA[0001] ❌ Cannot read property 'bucket1user01#kqbuft7c' of undefined 

# noobaa bucket create temp
INFO[0001] ✅ Exists: NooBaa "noobaa"                    
INFO[0001] ✅ Exists: Service "noobaa-mgmt"              
INFO[0001] ✅ Exists: Secret "noobaa-operator"           
INFO[0001] ✅ Exists: Secret "noobaa-admin"              
INFO[0001] ✈️  RPC: bucket.create_bucket() Request: {Name:temp Tiering: BucketClaim:<nil> Namespace:<nil>} 
WARN[0001] RPC: GetConnection creating connection to wss://localhost:40653/rpc/ 0xc000437b30 
INFO[0001] RPC: Connecting websocket (0xc000437b30) &{RPC:0xc0004366e0 Address:wss://localhost:40653/rpc/ State:init WS:<nil> PendingRequests:map[] NextRequestID:0 Lock:{state:1 sema:0} ReconnectDelay:0s} 
INFO[0001] RPC: Connected websocket (0xc000437b30) &{RPC:0xc0004366e0 Address:wss://localhost:40653/rpc/ State:init WS:<nil> PendingRequests:map[] NextRequestID:0 Lock:{state:1 sema:0} ReconnectDelay:0s} 
ERRO[0001] ⚠️  RPC: bucket.create_bucket() Response Error: Code=INTERNAL Message=Cannot read property 'temp#kqbwt6y7' of undefined 
FATA[0001] Cannot read property 'temp#kqbwt6y7' of undefined 

noobaa-core pod logs:

Jun-25 4:50:18.380 [WebServer/36] [ERROR] CONSOLE:: RPC._on_request: ERROR srv bucket_api.create_bucket reqid wss://localhost:33227/rpc/-0 connid ws://[::ffff:127.0.0.1]:47566/(eg0xy113e) TypeError: Cannot read property 'bucket1user01#kqbuvpfv' of undefined
    at Object.check_tier_exists (/root/node_modules/noobaa-core/src/server/system_services/tier_server.js:864:33)
    at Object.create_bucket (/root/node_modules/noobaa-core/src/server/system_services/bucket_server.js:129:21)
    at Object.server_func (/root/node_modules/noobaa-core/src/rpc/rpc.js:113:48)
    at /root/node_modules/noobaa-core/src/rpc/rpc.js:340:46
    at RPC._on_request (/root/node_modules/noobaa-core/src/rpc/rpc.js:345:19)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
```

--- a/src/server/system_services/bucket_server.js
+++ b/src/server/system_services/bucket_server.js
@@ -164,11 +164,11 @@ async function create_bucket(req) {
     if (req.rpc_params.namespace) {
         const read_resources = _.compact(req.rpc_params.namespace.read_resources
             .map(nsr => {
-                const res = req.system.namespace_resources_by_name[nsr.resource];
+                const res = req.system.namespace_resources_by_name && req.system.namespace_resources_by_name[nsr.resource];
                 return res && { resource: res._id, path: nsr.path };
             })
         );
-        const wr_obj = req.rpc_params.namespace.write_resource &&
+        const wr_obj = req.rpc_params.namespace.write_resource && req.system.namespace_resources_by_name &&
             req.system.namespace_resources_by_name[req.rpc_params.namespace.write_resource.resource];
         const write_resource = wr_obj && { resource: wr_obj._id, path: req.rpc_params.namespace.write_resource.path };
         if (req.rpc_params.namespace.read_resources &&
@@ -604,14 +604,15 @@ function get_bucket_changes_namespace(req, bucket, update_request, single_bucket
 
     const read_resources = _.compact(update_request.namespace.read_resources
         .map(nsr => {
-            const res = req.system.namespace_resources_by_name[nsr.resource];
+            const res = req.system.namespace_resources_by_name && req.system.namespace_resources_by_name[nsr.resource];
             return res && { resource: res._id, path: nsr.path };
         }));
     if (!read_resources.length || (read_resources.length !== update_request.namespace.read_resources.length)) {
         throw new RpcError('INVALID_READ_RESOURCES');
     }
     _.set(single_bucket_update, 'namespace.read_resources', read_resources);
-    const wr_obj = req.system.namespace_resources_by_name[update_request.namespace.write_resource.resource];
+    const wr_obj = req.system.namespace_resources_by_name &&
+        req.system.namespace_resources_by_name[update_request.namespace.write_resource.resource];
     const write_resource = wr_obj && { resource: wr_obj._id, path: update_request.namespace.write_resource.path };
     if (!write_resource) throw new RpcError('INVALID_WRITE_RESOURCES');
     _.set(single_bucket_update, 'namespace.write_resource', write_resource);
@@ -1379,7 +1380,7 @@ function validate_bucket_creation(req) {
 
 function validate_nsfs_bucket(req) {
     // do not allow creation of 2 nsfs buckets on the same path - RPC 
-    if (req.rpc_params.namespace) {
+    if (req.rpc_params.namespace && req.system.namespace_resources_by_name) {
         const write_resource = req.system.namespace_resources_by_name[req.rpc_params.namespace.write_resource.resource];
         const bucket_nsfs_config = write_resource && write_resource.nsfs_config;
         if (bucket_nsfs_config) {

--- a/src/server/system_services/tier_server.js
+++ b/src/server/system_services/tier_server.js
@@ -596,7 +596,7 @@ function find_bucket_by_tier(req) {
 
 function find_tier_by_name(req, tier_name) {
     const name = tier_name || req.rpc_params.name;
-    const tier = req.system.tiers_by_name[name.unwrap()];
+    const tier = req.system.tiers_by_name && req.system.tiers_by_name[name.unwrap()];
     if (!tier) {
         throw new RpcError('NO_SUCH_TIER', 'No such tier: ' + name);
     }
@@ -605,7 +605,7 @@ function find_tier_by_name(req, tier_name) {
 
 function find_policy_by_name(req) {
     const name = req.rpc_params.name;
-    const policy = req.system.tiering_policies_by_name[name.unwrap()];
+    const policy = req.system.tiering_policies_by_name && req.system.tiering_policies_by_name[name.unwrap()];
     if (!policy) {
         throw new RpcError('NO_SUCH_TIERING_POLICY', 'No such tiering policy: ' + name);
     }
@@ -856,12 +856,12 @@ function throw_on_invalid_pools_for_tier(pools) {
 
 function check_tiering_policy_exists(req, policy_name) {
     const name = policy_name instanceof SensitiveString ? policy_name.unwrap() : policy_name;
-    if (req.system.tiering_policies_by_name[name]) throw new RpcError(`Policy ${policy_name} already exists`);
+    if (req.system.tiering_policies_by_name && req.system.tiering_policies_by_name[name]) throw new RpcError(`Policy ${policy_name} already exists`);
 }
 
 function check_tier_exists(req, tier_name) {
     const name = tier_name instanceof SensitiveString ? tier_name.unwrap() : tier_name;
-    if (req.system.tiers_by_name[name]) throw new RpcError(`Tier ${tier_name} already exists`);
+    if (req.system.tiers_by_name && req.system.tiers_by_name[name]) throw new RpcError(`Tier ${tier_name} already exists`);
 }
 
 

